OpenEdXMobile/build.gradle

```diff
@@ -144,8 +144,8 @@ dependencies {
     implementation 'com.google.android.exoplayer:exoplayer:2.16.0'
 
     // Google cast sdk
-    implementation "androidx.mediarouter:mediarouter:$androidXVersion"
-    implementation 'com.google.android.gms:play-services-cast-framework:16.2.0'
+    implementation "androidx.mediarouter:mediarouter:1.3.1"
+    implementation 'com.google.android.gms:play-services-cast-framework:21.2.0'
```

Comment:
Updating the libraries has increased some warnings in some classes: 
- GoogleCastDelegate
- CastOptionsProvider

Please resolve these and others (if you can find them)as well

Reply:
Most of the warnings are removed. 
`CastContext.getSharedInstance(this.context)` method is deprecated, I didn't fix it because of the unavailability of the casting device.

Reply:
Awesome but `getSharedInstance ` it's deprecated, we should update and test it with this block: 
```
castContext = CastContext.getSharedInstance(this.context, Executors.newSingleThreadExecutor()).getResult();
```

Reply:
There is also another warning in the file. We can probably update it like this: 
```
private void registerRemoteCallback() {
        if (castContext.getCastState() == CastState.CONNECTED) {
            RemoteMediaClient client = castSession.getRemoteMediaClient();
            if (client != null) {
                client.registerCallback(this);
            }
        }
    }
    ```

Reply:
We will fix this in code improvements